### PR TITLE
Add nessie to spark3/spark2 runtime jar

### PR DESCRIPTION
**Pull request by rymurr**
Resulting from discussion in #1887 this adds Nessie catalog to the spark runtimes. This adds ~496K to the 18M runtime jar.

Note: the extra dependency `'org.glassfish.jersey.media:jersey-media-json-jackson` is the only Nessie dependency that isn't already included in Spark. The version of this dependency is pinned to the specific version of jersey in Spark 2.4 and 3 to ensure compatibility in the shared jar.

This has been tested externally to ensure the catalog and all its dependencies are loaded correctly.

### CODE REVIEW

**Comment by jackye1995** [comment]
What about the Flink and Hive runtimes?

**Comment by rymurr** [comment]
> What about the Flink and Hive runtimes?

I haven't tested Flink or Hive yet so want to do those as separate PRs. Working on Hive today :-)

**Review by jackye1995 — APPROVED** [review]

**Comment by rdblue** [comment]
Adding new classes and dependencies into the runtime Jars requires an update to the LICENSE and NOTICE files for those Jars. Could you also update those in this PR, @rymurr?

@jacques-n, can you help with the license update validation?

**Comment by jacques-n** [comment]
Will do.

**Comment by rymurr** [comment]
Ive added Nessie and Jersey to `NOTICE` and `LICENSE`. Not 100% sure about Jersey as its CDDL licensed in Spark2. @jacques-n and @laurentgo look ok to you?

File: spark3-runtime/NOTICE
Comment (rdblue) [inline spark3-runtime/NOTICE:507]: I don't think that we typically need the copyright part, or acknowledgement that there is ASF software. Since this is boilerplate and the relevant information is in LICENSE, we can probably omit it.

File: spark3-runtime/LICENSE
Comment (rdblue) [inline spark3-runtime/LICENSE:596]: We need to include a copy of the license text here.

File: spark3-runtime/LICENSE
Comment (rdblue) [inline spark3-runtime/LICENSE:592]: EPL 2.0 is listed under [Category B, "weak" copyleft licenses](https://www.apache.org/legal/resolved.html). The relevant sections from that page are:

> Software under the following licenses may be included in binary form within an Apache product if the inclusion is appropriately labeled (see above)

> By including only the object/binary form, there is less exposed surface area of the third-party work from which a work might be derived; this addresses the second guiding principle of this policy.

And here's the part about labelling:

> APPROPRIATELY LABELLED CONDITION
> In all Category B cases our users should not be surprised at their inclusion in our products. By attaching an appropriate and prominent label to the distribution users are less likely to be unaware of restrictions significantly different from those of the Apache License. An appropriate and prominent label is a label the user will read while learning about the distribution - for example in a README, and it should identify the third-party product, its licensing, and a url to the its homepage. Please also ensure to comply with any attribution/notice requirements in the specific license in question.

I think that means we will need a "prominent label" that a user will read while learning about the distribution. I would add one as a new section to the project's README, as well as on the releases page. We should also link to it from anywhere we recommend downloading a runtime Jar on the site. We should also add any notes needed to comply with [EPL's requirements](https://www.eclipse.org/legal/epl-2.0/#requirements).

Is it possible to remove this dependency? That might be easier.

Comment (jacques-n) [reply spark3-runtime/LICENSE:592]: Somehow I apparently didn't post and lost the comments that I thought I had added to this ticket. @rymurr we should look at how other tools handle this. Jersey is used by most Apache projects and most of it is already packaged by tools like Spark, Hive, etc. The only thing we're actually doing here is adding the extension for jackson encoding/decoding for Jersey. 

@rdblue we're also exploring avoiding jersey entirely as Hive 2.x has a nasty old version of it that is causing us headaches.

Comment (rymurr) [reply spark3-runtime/LICENSE:592]: FWIW Spark distributes [LICENSE-binary](https://github.com/apache/spark/blob/master/LICENSE-binary) and [NOTICE-binary](https://github.com/apache/spark/blob/master/NOTICE-binary) with a footnote for any non-apache licenses.

Hopefully I can remove jersey entirely and skip EPL completely

**Comment by rdblue** [comment]
@rymurr or @jacques-n, any update on this? I would like to aim to get it into the next release.

**Comment by jacques-n** [comment]
We just merged [Nessie#575](https://github.com/projectnessie/nessie/pull/575) which removes the Jaxrs/jersey requirement entirely. We'll push a release shortly that updates the dependency to avoid this extra set of licenses.

**Comment by rymurr** [comment]
I have updated to remove jax-rs and correct the license and notice. This should be ready to merge once #2016 is merged

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks @rymurr! I merged this. Do you also want to create PRs for the Flink and Hive runtime Jars?

**Comment by rymurr** [comment]
Thanks @rdblue!

I am running through some flink Tests and wkll raise the relevant PR soon.

Hive is a bit more work as the discussion around properties is relevant to how we specify custom catalogs. Hope to have something for this soon too.

Happy new year!

**Pull request merged**
